src/backend/src/utils/reimbursement-requests.utils.ts

```diff
@@ -50,3 +58,73 @@ export const validateReimbursementProducts = async (reimbursementProductCreateAr
     throw new HttpException(400, `The following projects or work packages do not exist: ${missingWbsNumbers.join(', ')}`);
   }
 };
+
+/**
+ * This function updates the current reimbursement products of a reimbursement request versus the new reimbursement products of a reimbursement request.
+ * @param currentReimbursementProducts the current reimbursement products of a reimbursement request
+ * @param updatedReimbursementProducts the new reimbursement products to compare
+ * @param reimbursementRequestId the reimbursement request that is being changed id
+ */
+export const updateReimbursementProducts = async (
+  currentReimbursementProducts: Reimbursement_Product[],
+  updatedReimbursementProducts: ReimbursementProductCreateArgs[],
+  reimbursementRequestId: string
+) => {
+  //if a product has an id that means it existed before and was updated
+  const updatedProducts = updatedReimbursementProducts.filter((product) => product.id);
+
+  //Check to make sure that the updated products actually exist in the database
+  const prismaProductIds = currentReimbursementProducts.map((product) => product.reimbursementProductId);
```

Comment:
ok I see why we need all this, but we can at minimum make it more readable. Let's start by putting lines 76-86 in a helper method called checkMissingIds or smthn, and have it take in the updatedProducts.

Reply:
also these variable names are hella similar, can we call updatedProducts existingProducts

Reply:
existing Products is the same as current products, how about updatedExistingProducts

Reply:
Sure